sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.appcenter.utils;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.os.Handler;
+
+import com.microsoft.appcenter.AbstractAppCenterTest;
+import com.microsoft.appcenter.AppCenterService;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(PowerMockRunner.class)
+public class ApplicationLifecycleListenerTest {
+
+    @Mock
+    Handler mHandlerMock;
+
+    @Mock
+    Activity mActivityMock;
+
+    private ApplicationLifecycleListener mApplicationLifecycleListener;
+
+    @Before
+    public void setUp() {
+        mApplicationLifecycleListener = new ApplicationLifecycleListener(mHandlerMock);
+        doAnswer(new Answer<Void>() {
+
+            @Override
+            public Void answer(InvocationOnMock invocation) {
+                ((Runnable) invocation.getArguments()[0]).run();
+                return null;
+            }
+        }).when(mHandlerMock).postDelayed(any(Runnable.class), anyLong());
+        doAnswer(new Answer<Void>() {
+
+            @Override
+            public Void answer(InvocationOnMock invocation) {
+                ((Runnable) invocation.getArguments()[0]).run();
```

Comment:
Why executing the runnable since here the goal is to mock canceling it?

Reply:
Looks like we don't need to mock anything for removeCallbacks, we just verify invocations.